### PR TITLE
Keep redrawing until video decoder is done decoding requested frame

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* bits and pieces for https://github.com/rerun-io/rerun/issues/7373
    * but far from having that solved

Video decoder forwards now more rich status information upon decode.
This is useful for error handling and - the immediate motivation - to keep redrawing while we're waiting on pending frames.

Before, when clicking somewhere on the timeline, we wouldn't redraw until the mouse moves, which for back-seeking (and far forward-seeking) usually means that we don't get the new frame until the decoder catches up.

I also took the liberty to move the video re_renderer module around a bit - it was falsely sorted under `renderer` which are all about renderers that process draw data and get submitted to a view builder (which is not the case here!).

https://github.com/user-attachments/assets/623d712b-384b-4ffa-bcbf-3a80885ccef6

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7398?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7398?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7398)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_renderer/src/video/decoder/web.rs [hunk 1]
```diff
@@ -268,6 +263,7 @@ impl VideoDecoder {
         chunk.set_duration(sample.duration.as_f64());
         let Some(chunk) = EncodedVideoChunk::new(&chunk)
             .inspect_err(|err| {
+                // TODO(#7373): return this error once the decoder tries to return a frame for this sample. how exactly?
```
Comment (jprochazk) [inline crates/viewer/re_renderer/src/video/decoder/web.rs:266]: We may have to set the decoder into an "error state" if it fails to decode a frame, as there may not be a reasonable way to continue decoding. In that case we would just store the error, and close the underlying decoder.

The only reason why decoding should fail is if their video is incorrectly encoded or corrupted in some way. We could for example try again from the next keyframe, as long as the user requests it. So we'd store the _segment_ which is broken inside the error state, alongside the error message from the decoder, and present that to the user until they seek to a different segment.

Comment (Wumpf) [reply crates/viewer/re_renderer/src/video/decoder/web.rs:266]: makes sense. I suppose until we have a concrete example of a video that runs into this case to experiment with we shouldn't go overboard with recovery strategies as we don't know how they will work out. But marking segments as broken seems like a good strategy 👍 :)

Comment (jprochazk) [reply crates/viewer/re_renderer/src/video/decoder/web.rs:266]: I don't know if you've experienced this scenario: You're playing a video on the web, and it suddenly fails with an error of some kind. The player just stops, and the controls become inactive, so you can't seek to a different point in the video anymore. But then you reload the page, and seek after the point where the player previously stopped with an error, and the video continues playing back just fine.

I think we could do a lot better than web video players here, by always allowing seek even in the case of an error, and more importantly also highlight which segments are broken on the timeline, so the user knows exactly where to seek next to try and catch a part of the video which _can_ be played back.

This kind of behavior would definitely follow our "handle anything the user throws at it" philosophy

**Review by jprochazk — APPROVED** [review]
Nice!

File: crates/viewer/re_renderer/src/video/decoder/web.rs (hunk 1 above)
Comment (jprochazk) [reply crates/viewer/re_renderer/src/video/decoder/web.rs:266]: (actually, I will turn this into a comment on the existing issue)

**Pull request merged**